meson.build

```diff
@@ -56,10 +57,20 @@ taisei_c_args = cc.get_supported_arguments(
     '-Wstrict-prototypes',
     '-Wtype-limits',
     '-Wunneeded-internal-declaration',
-    '-Wunreachable-code',
     '-Wunreachable-code-loop-increment',
 )
 
+if cc.compiles('''
+int main() {
+    return 0;
+    __builtin_unreachable();
+}
+''', name : 'unreachable __builtin_unreachable() with -Werror', args : [ '-Wunreachable-code', '-Werror' ] )
+    taisei_c_args += cc.get_supported_arguments('-Wunreachable-code')
+else
+    warning('-Wunreachable-code will not be used due to incorrect handling of __builtin_unreachable')
+endif
+
```

Comment:
I'm not sure this check is necessary. I think it's fine to let buggy compilers spit bogus warnings, so long as they are harmless. Does the Elbrus compiler suffer from this issue as well?

Reply:
Yes, it was caused by me misunderstanding cause of some warnings, and after commit 36493d3, it is no more needed.